docs/designers-developers/developers/feature-flags.md

```diff
@@ -0,0 +1,114 @@
+
+If you write the following code:
+```js
+if ( window.GUTENBERG_PHASE === 2 ) {
+	phaseTwoFeature();
+}
+```
+
+When building the codebase for the plugin the variable will be replaced with the number literal `2`:
+```js
+if ( 2 === 2 ) {
+	phaseTwoFeature();
+}
+```
+
+Any code within the body of the if statement will be executed within the gutenberg plugin since `2 === 2` evaluates to `true`.
+
+For core, the `window.GUTENBERG_PHASE` variable is replaced with `1`, so the built code will look like:
+```js
+if ( 1 === 2 ) {
+	phaseTwoFeature();
+}
+```
+
+`1 === 2` evaluates to false so the phase 2 feature will not be executed within core.
+
+### Dead Code Elimination
+
+When building code for production, webpack 'minifies' code (https://en.wikipedia.org/wiki/Minification_(programming)), removing the amount of unnecessary JavaScript as much as possible. One of the steps involves something known as 'dead code elimination'. 
+
+When the following code is encountered, webpack determines that the surrounding `if`statement is unnecessary:
+```js
+if ( 2 === 2 ) {
+	phaseTwoFeature();
+}
+```
+
+ The condition will alway evaluates to `true`, so can be removed leaving just the code in the body:
+ ```js
+ phaseTwoFeature();
+ ```
+
+Similarly when building for core, the condition in the following `if` statement always resolves to false:
+```js
+if ( 1 === 2 ) {
+	phaseTwoFeature();
+}
+```
+
+The minification process will remove the entire `if` statement including the body, ensuring code destined for phase 2 is not included in the built JavaScript intended for core.
+
+## Basic Use
+
+A phase 2 function or constant should be exported using the following ternary syntax:
+
+```js
+function myPhaseTwoFeature() {
+	// implementation
+}
+
+export const phaseTwoFeature = window.GUTENBERG_PHASE === 2 ? myPhaseTwoFeature : undefined;
+```
+
+In phase 1 environments the `phaseTwoFeature` export will be `undefined`.
+
+If you're attempting to import and call a phase 2 feature, be sure to wrap the call to the function in an if statement to avoid an error:
+```js
+import { phaseTwoFeature } from '@wordpress/foo';
+
+if ( window.GUTENBERG_PHASE === 2) {
+	phaseTwoFeature();
+}
+```
+
+## FAQ
+
+#### Since `GUTENBERG_PHASE` is a global object, why do I have to access it through the `window` object, e.g. `window.GUTENBERG_PHASE`?
+
+Third-party consumers of WordPress npm packages may have their own build process, resulting in `window.GUTENBERG_PHASE` not being replaced at build time with the appropriate number. In JavaScript, when attempting to access an undefined global variable an error is thrown, a situation that would cause lots of issues for third-parties (or additional build configuration). Accessing the `GUTENBERG_PHASE` variable through the `window` object circumvents this, the value of the variable evaluates to `undefined` resulting in WordPress packages continuing to work for third-parties.
```

Comment:
This also hints at another problem though: They might have their own build process, so `window` can't be assumed to be defined either.

Reply:
Hmm, hadn't considered that. I suppose that would be usage outside the browser. 🤔 

It's a shame that the condition might end up becoming quite verbose. I'd wanted to keep it as sort and simple as possible.